backend/app/services/knowledge/summary_service.py

```
@@ -0,0 +1,786 @@
+            flag_modified(document, "summary")
+            self.db.commit()
+
+            logger.info(
+                f"[SummaryService] Document summary status set to generating: "
+                f"document_id={document_id}"
+            )
+
+            # 7. Get document content
+            content = await self._get_document_content(document)
+            if not content:
+                raise Exception("Failed to get document content")
```

Ruff (0.14.11)

Abstract `raise` to an inner function

(TRY301)